### PR TITLE
feat: update create event form, and save event to db

### PR DESCRIPTION
**Pull request by pointmekin**
# Notes
- Add online event support
  - Allow user to select between online and offline event via a toggle button
- Save event to MongoDB
- Add event summary page 
- Fix Yup validation schema
- Fix Types
- Fix bugs

### CODE REVIEW

**Comment by pointmekin** [comment]
Rebased

File: components/event/CreateEvent/Steps/CreateEventStep3.tsx [hunk 1]
```diff
@@ -171,6 +188,7 @@ function CreateEventStep3() {
             </FormHelperText>
           )}
         </FormControl>
+        <div>{JSON.stringify(errors)}</div>
```
Comment (chuwong35122) [inline components/event/CreateEvent/Steps/CreateEventStep3.tsx:191]: yo

File: components/event/MyEvents/EventsList.tsx [hunk 2]
```diff
@@ -3,14 +3,32 @@ import {
...
+
+  useEffect(() => {
+    getEventFromOrganizer()
+  }, [])
+
+  const handleClickEvent = (eventId: string | undefined) => {
+    if (!eventId) return
+    router.push({
+      pathname: `/event/detail/${eventId}`
+    })
```
Comment (chuwong35122) [inline components/event/MyEvents/EventsList.tsx:29]:
```
if(eventId) {
router.push(...)
}
```

File: components/event/MyEvents/EventsList.tsx [hunk 3]
```diff
@@ -20,29 +38,49 @@ function EventsList() {
...
+            >
+              <Card
+                sx={{ p: 0, height: 400, width: "100%", borderRadius: 0 }}
+                key={index}
+                elevation={0}
+              >
+                <CardMedia
+                  sx={{ height: 140 }}
+                  image={event.image as string}
+                  title="green iguana"
```
Comment (chuwong35122) [inline components/event/MyEvents/EventsList.tsx:56]: change title

File: hooks/useEvents.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+export const useEvents = () => {
+  const [events, setEvents] = useState<Event[]>([])
+  const [currentEvent, setCurrentEvent] = useState<Event>()
+
+  const getEventFromOrganizer = async () => {
+    try {
+      const res = await client.get<Event[]>("/event/getEventFromOrganizer")
+      const events: Event[] = res.data
+      setEvents(events)
```
Comment (chuwong35122) [inline hooks/useEvents.ts:13]: can you also add
`return events`
too?

File: components/UI/button/ContainedButton.tsx [hunk 5]
```diff
@@ -44,6 +46,7 @@ function ContainedButton({
       <Button
         type={type ?? "button"}
         onClick={(e?: any) => {
+          if (isLink) return null
```
[the remark this answers is not in the document]
Comment (pointmekin) [reply components/UI/button/ContainedButton.tsx:49]: This one is weird. When wrapping next link around a ContainedButton, the link will not work, unless the button doesnt have an onclick.

any ideas to improve this?

File: components/event/CreateEvent/Steps/CreateEventStep3.tsx (hunk 1 above)
Comment (pointmekin) [reply components/event/CreateEvent/Steps/CreateEventStep3.tsx:191]: Lmao

File: components/event/MyEvents/EventsList.tsx (hunk 2 above)
Comment (pointmekin) [reply components/event/MyEvents/EventsList.tsx:29]: They are the same tho

File: components/event/MyEvents/EventsList.tsx (hunk 3 above)
Comment (pointmekin) [reply components/event/MyEvents/EventsList.tsx:56]: Ahh.. lmao

File: hooks/useEvents.ts (hunk 4 above)
Comment (pointmekin) [reply hooks/useEvents.ts:13]: Ok

File: components/event/MyEvents/EventsList.tsx [hunk 6]
```diff
@@ -3,14 +3,32 @@ import {
...
+import { useEvents } from "../../../hooks/useEvents"
+import HoverCard from "../../motion/HoverCard"
+import { Event } from "../../../interfaces/event/event.interface"
+import { useRouter } from "next/router"
 
 function EventsList() {
+  const { events, getEventFromOrganizer } = useEvents()
+  const router = useRouter()
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (pointmekin) [reply components/event/MyEvents/EventsList.tsx:21]: If you call other async functions inside a use effect, you dont need async here.

however, if you want to pass an async to use effect directly, then yea it gotta be async

**Pull request merged**